instrumentation/influxdb-2.4/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/influxdb/v2_4/InfluxDbClientTest.java

```diff
@@ -0,0 +1,311 @@
+import static io.opentelemetry.sdk.testing.assertj.OpenTelemetryAssertions.equalTo;
+import static io.opentelemetry.semconv.SemanticAttributes.DB_CONNECTION_STRING;
+import static io.opentelemetry.semconv.SemanticAttributes.DB_NAME;
+import static io.opentelemetry.semconv.SemanticAttributes.DB_OPERATION;
+import static io.opentelemetry.semconv.SemanticAttributes.DB_STATEMENT;
+import static io.opentelemetry.semconv.SemanticAttributes.DB_SYSTEM;
+import static io.opentelemetry.semconv.SemanticAttributes.NETWORK_TYPE;
+import static java.util.Arrays.asList;
+
+import io.opentelemetry.api.trace.SpanKind;
+import io.opentelemetry.instrumentation.testing.junit.AgentInstrumentationExtension;
+import io.opentelemetry.instrumentation.testing.junit.InstrumentationExtension;
+import io.opentelemetry.sdk.testing.assertj.AttributeAssertion;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.LinkedBlockingQueue;
+import java.util.concurrent.TimeUnit;
+import org.influxdb.InfluxDB;
+import org.influxdb.InfluxDBFactory;
+import org.influxdb.dto.BatchPoints;
+import org.influxdb.dto.Point;
+import org.influxdb.dto.Query;
+import org.influxdb.dto.QueryResult;
+import org.junit.Assert;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestInstance;
+import org.junit.jupiter.api.TestInstance.Lifecycle;
+import org.junit.jupiter.api.extension.RegisterExtension;
+import org.testcontainers.containers.GenericContainer;
+
+// ignore @Deprecated annotation warning.
+@SuppressWarnings("deprecation")
+@TestInstance(Lifecycle.PER_CLASS)
+class InfluxDbClientTest {
+
+  @RegisterExtension
+  private static final InstrumentationExtension testing = AgentInstrumentationExtension.create();
+
+  private static final GenericContainer<?> influxDbServer =
+      new GenericContainer<>("influxdb:1.8.10-alpine").withExposedPorts(8086);
+
+  private static InfluxDB influxDb;
+
+  private static final String databaseName = "mydb";
+
+  private static String serverURL;
+
+  private static int port;
+
+  @BeforeAll
+  void setup() {
+    influxDbServer.start();
+    port = influxDbServer.getMappedPort(8086);
+    String host = influxDbServer.getHost();
+    serverURL = "http://" + host + ":" + port + "/";
+    String username = "root";
+    String password = "root";
+    influxDb = InfluxDBFactory.connect(serverURL, username, password);
+    influxDb.createDatabase(databaseName);
+  }
+
+  @AfterAll
+  void cleanup() {
+    influxDb.deleteDatabase(databaseName);
+    influxDb.close();
+  }
+
+  @BeforeEach
+  void reset() {
+    testing.clearData();
+  }
+
+  @Test
+  void testQueryAndModifyWithOneArgument() {
+    String dbName = databaseName + System.currentTimeMillis();
```

Comment:
If you need to create only one extra database you could use `String dbName = "mydb2";` if you need many just use an `AtomicInteger` to create sequential numbers to append to the base database name.